pennylane/pulse/rydberg_hamiltonian.py

```diff
@@ -0,0 +1,289 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This module contains the classes/functions needed to simulate the evolution of ensembles of
+individual (trapped) rydberg atoms under the excitation of several laser fields."""
+import warnings
+from dataclasses import dataclass
+from typing import Callable, List, Union
+
+import numpy as np
+
+import pennylane as qml
+from pennylane.ops import SProd
+from pennylane.wires import Wires
+
+from .parametrized_hamiltonian import ParametrizedHamiltonian
+
+
+def rydberg_interaction(register: list, wires=None, interaction_coeff: float = 862690 * 2 * np.pi):
+    r"""Returns a :class:`ParametrizedHamiltonian` representing the interaction of an ensemble of
+    Rydberg atoms due to the Rydberg blockade
+
+    .. math::
+
+        \sum_{i<j} V_{ij} n_i n_j
+
+    where :math:`n_i` corresponds to the projector on the Rydberg state of the atom :math:`i`, and
+    :math:`V_{ij}` is the van der Waals potential:
+
+    .. math::
+
+        V_{ij} = \frac{C_6}{R_{ij}^6}
+
+    where :math:`R_{ij}` is the distance between the atoms :math:`i` and :math:`j`, and :math:`C_6`
+    is the Rydberg interaction constant, which defaults to :math:`862690 \times 2\pi MHz \times \mu m^6`.
```

Comment:
I definitely regularly convert unit prefixes to 10^whatever, let them cancel out and move things around, and get units at the end, if that's what you mean. Just have to make sure everything else in the equation (in this case R) uses the same units. So we expect `coordinates` expressed in um from some origin, to match the units on C6. 

But I guess we need to consider that the overall coefficient comes out in MHz then, and whether that lines up with the other coefficients.

If that's not what you meant, I'm not sure what you mean, please clarify 😄

Reply:
Here I really just meant the documentation and whether it is common to have MHz um written explicitly, i.e. whether this is something taken from somewhere else or something we do to indicate the orders of magnitude for frequencies/energies and distances

Reply:
Right! I don't think its super standard, I think its more something we are doing to avoid things like 10e-24 being fed into the ODE solver. But I'm not sure, maybe Albert saw it expressed this way somewhere. I have only seen 5.42e-24 rad/s m, but it comes out to the same.

Reply:
I'm not sure. Albert added this documentation, and I haven't been able to find the value of the Rydberg interaction constant used here online.

Reply:
Actually on second thought I think the factor of $2\pi$ seems off? 

The QuEra constant is $5.42 \cdot 10^{-24}$ rad/s $m^6$. 1 Hz = $2\pi$ rad/s

$5.42 \cdot 10^{-24}$ rad/s $m^6$ 
= $5.42 \cdot 10^{12}$ rad/s $\mu m^6$ 
= $5.42 \cdot 10^{12} / 2\pi$ Hz $\mu m^6$ 
= $8.63 \cdot 10^{11}$ Hz $\mu m^6$ 
= $863000$ MHz $\mu m^6$

Probably whatever number we end on we should also include a source.

Reply:
I vaguely recall seeing it somewhere, possibly in the Pulser documentation, and thinking 'Oh, that's where he got it'. But I'm not sure where.

Reply:
So the interaction coefficient he specified is actually in rad/s um^6? I won't change anything for now so that we can find a concrete source to refer to before changes are made.

Reply:
Yes, and definitely dont suggest users to put coordinates in `[5e-6, 5e-6]` that will be a numerical nightmare 😅 We should have all units such that we operate with on unit magnitude (every concrete numerical value is $\mathcal{O}(1)$

Reply:
I've updated the documentation to not include the 2 * pi term in the interaction coefficient, as well as changed the default value of `interaction_coeff` to 862690. With MHz um^6 units the units should still be correct, and the actual Hamiltonian should be correct now as well.